torch_frame/gbdt/gbdt.py

```diff
@@ -81,41 +107,36 @@ def predict(self, tf_test: TensorFrame) -> Tensor:
         return pred
 
     @property
-    def metric(self) -> str:
-        r"""Metric to compute for different tasks. root mean squared error for
-        regression, ROC-AUC for binary classification, and accuracy for
-        multi-label classification task.
-        """
-        if self.task_type == TaskType.REGRESSION:
-            return 'rmse'
-        elif self.task_type == TaskType.BINARY_CLASSIFICATION:
-            return 'rocauc'
-        elif self.task_type == TaskType.MULTICLASS_CLASSIFICATION:
-            return 'acc'
-        else:
-            raise ValueError(f"metric is not defined for {self.task_type}")
+    def metric(self) -> Metric:
+        r"""Evaluation metric to tune."""
+        return self._metric
```

Comment:
Is there a need for this property? Why not just set `self.metric` in `__init__`?

Reply:
Good point. Deleted.